scripts/swap-beacon/src/index.ts

```diff
@@ -0,0 +1,308 @@
+            val.tokensToReceive.reduce(
+              (acc, val) => BigInt(acc + val.amount),
+              0n
+            )
+        ),
+      0n
+    );
+
+    if (buyToken <= sellToken) {
+      console.log(
+        "Sell token is greater than buy token for opportunity:",
+        opportunity
+      );
+      console.log("Sell token:", sellToken.toString());
+      console.log("Buy token:", buyToken.toString());
+      console.log("Swaps sell:", JSON.stringify(swapsSell));
+      console.log("Swaps buy:", JSON.stringify(swapsBuy));
+      return undefined;
+    }
+
+    const sellTokens = [
+      {
+        token: base,
+        amount: sellToken,
+      },
+    ];
+    const buyTokens = [
+      {
+        token: base,
+        amount: buyToken,
+      },
+    ];
+
+    const targetCalldata = this.makeMulticallCalldata(
+      opportunity,
+      swapsSell,
+      swapsBuy,
+      sellTokens,
+      buyTokens
+    );
+
+    return {
+      ...opportunity,
+      targetContract,
+      targetCalldata,
+      targetCallValue,
+      sellTokens,
+      buyTokens,
+    };
+  }
+
+  async convertOpportunity(
+    opportunity: Opportunity,
+    base: Address
+  ): Promise<Opportunity | undefined> {
+    const optimalAdaptersSell = opportunity.sellTokens.map((sellToken) =>
+      this.getOptimalAdapter(base, sellToken.token)
+    );
+    const optimalAdaptersBuy = opportunity.buyTokens.map((buyToken) =>
+      this.getOptimalAdapter(buyToken.token, base)
+    );
+
+    const swapsSell = (
+      await Promise.all(
+        optimalAdaptersSell.map(async (adapter, index) => {
+          if (adapter === undefined) {
+            return [];
+          }
+          return adapter.constructSwaps(
+            opportunity.chainId,
+            base,
+            opportunity.sellTokens[index].token,
+            undefined,
+            opportunity.sellTokens[index].amount
+          );
+        })
+      )
+    ).reduce((acc, val) => acc.concat(val), []);
+    const swapsBuy = (
+      await Promise.all(
+        optimalAdaptersBuy.map(async (adapter, index) => {
+          if (adapter === undefined) {
+            return [];
+          }
+          return adapter.constructSwaps(
+            opportunity.chainId,
+            opportunity.buyTokens[index].token,
+            base,
+            opportunity.buyTokens[index].amount,
+            undefined
+          );
+        })
+      )
+    ).reduce((acc, val) => acc.concat(val), []);
+
+    return this.createSwapOpportunity(opportunity, base, swapsSell, swapsBuy);
+  }
+
+  async opportunityHandler(opportunity: Opportunity) {
+    const swapAdapterConfig = getSwapAdapterConfig(opportunity.chainId);
+
+    if (
+      opportunity.targetContract.toLowerCase() ===
+      swapAdapterConfig.multicallAdapter.toLowerCase()
+    ) {
+      return;
+    }
+
+    await Promise.all(
+      swapAdapterConfig.liquidAssets.map(async (base) => {
+        const convertedOpportunity = await this.convertOpportunity(
```

Comment:
does a single error here cause the whole script to exit? Is this a good behavior?